### ISSUE TITLE
bug(input, select, autocomplete): wrong cursor on hover inner padding

### ISSUE THREAD

**Issue by RobinKamps** [angular/components#26491]
### Is this a regression?

- [ ] Yes, this behavior used to work in the previous version

### Description

According to the Material Standard the cursor should always be text for autocomplete & input and pointer for select.
Instead the cursor is only set to text/pointer on the standard html element - not the container, which has a padding where the cursors are default.
On the select it gets a bit more wierd, because the cursor wobbles from default (on th container padding) to text (on th label i guess) to the pointer (on th html standard input).
![inputs-cursor](https://user-images.githubusercontent.com/27373333/214418527-f8902de4-4fcd-455d-9a40-ea85ba811a4e.gif)

### Reproduction

Steps to reproduce:
1. head to the docs for autocomplete/input or select .e.g. https://material.angular.io/components/select/overview
2. hover over the elements

### Expected Behavior

the cursor should stay the same if hovered over the component.

### Actual Behavior

the cursor is different on the container padding / label (not material strandard compliant)

### Environment

- Angular: 15.1.1
- CDK/Material: 15.1.1
- Browser(s): all
- Operating System (e.g. Windows, macOS, Ubuntu): all

**Comment by CramericaIndustries** [comment]
Just was about to report the same bug. Wrong cursor appears also when you hover the mouse above the `mat-label` (red frame) 
![image](https://user-images.githubusercontent.com/5860703/224316918-71a7d2cf-4aa3-4a47-97bd-9f31eb6d47ab.png)

My CSS workaround for `mat-select` is...

```css
mat-form-field:has(mat-select) mat-label {
    cursor: pointer;
}
```

**Comment by felhag** [comment]
Looks to me like [this fix](https://github.com/angular/components/pull/26110) is the culprit. If you remove the `pointer-events: all` css property the cursor behaves fine.

**Comment by RobinKamps** [comment]
unfortunetaly this fix in 16.2.5 does not work on inputs and thus do not work on autocomplete (works on selects only).

![input-cursor](https://github.com/angular/components/assets/27373333/23bbf689-d6d9-4870-97b3-e4970a42d4d1)

**Comment by crisbeto** [comment]
Yeah I think it makes sense that it's styled as a text cursor since it is a text input.

**Comment by RobinKamps** [comment]
its not about the text cursor, but about where it should be applied.
the m2 material specs seems to have slighly different text cursor behavior. The labels have default cursor instead of text cursor on both appearances. On appearance outline the cursor is text for the wole input (no default cursor above and below the input) except for the right and left padding. In m3 / web material specs nothing can be found about the hover / cursor.

Angular Material MDC implementation has text cursor on the labels and different margins on appearance outline.

**Comment by RobinKamps** [comment]
![Animation](https://github.com/angular/components/assets/27373333/39c346d4-3caf-4610-b691-5d15007100c1)

**Comment by RobinKamps** [comment]
but anyway: using the whole input (without any default cursor) 

´´´
.mdc-text-field {
  cursor: text;
}
```

results in a much better user experience in my opinion then the different cursors/paddings etc.
